src/types/any.rs

```diff
@@ -194,7 +194,70 @@ impl PyAny {
         }
     }
 
-    /// Determines whether this object is callable.
+    /// Determines whether this object appears callable.
+    ///
+    /// This is equivalent to Python's [`callable()`][1] function.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use pyo3::prelude::*;
+    ///
+    /// # fn main() {
+    /// Python::with_gil(|py| {
+    ///        let builtins = PyModule::import(py, "builtins").unwrap();
+    ///        let print = builtins.getattr("print").unwrap();
+    ///        assert!(print.is_callable());
+    /// });
+    /// # }
+    /// ```
+    ///
+    /// This is equivalent to the Python expression `assert callable(print)`.
+    ///
+    /// # False positives
+    ///
+    /// Note that it is possible for this function to return `true`
+    /// for an object that can't be successfully called:
+    ///
+    /// ```rust
+    /// use pyo3::prelude::*;
+    ///
+    /// #[pyclass]
+    /// struct Foo {/* fields omitted */}
+    ///
+    /// # fn main() {
+    /// Python::with_gil(|py| {
+    ///     let my_module = PyModule::new(py, "my_module").unwrap();
+    ///     my_module.add_class::<Foo>();
+    ///     let foo = my_module.getattr("Foo").unwrap();
+    ///
+    ///     // Foo appears callable
+    ///     assert!(foo.is_callable());
+    ///
+    ///     // However, calling Foo will always return an error.
+    ///     assert!(foo.call0().is_err());
```

Comment:
I think this risks suggesting that this only happens with classes that don't define constructors. So I'd add as well that this is just *one* example of where this can happen.

There are other instances where it can happen, for example classes can sabotage their call slot:

```python
class Foo:
    __call__ = 42
```

Reply:
In other words: you can only check if you can call it, but not if you can call it successfully, with which arguments or in which environment.

This is why this check is only useful for such dynamically typed APIs with decisions based on "what kind of argument am I given here".

Reply:
> All these fine details are in the end irrelevant.

I agree. They are far too specific anyway. I've gutted it but I would like to keep the part about handling calling errors, since that seems to be what people want to do when they reach for `callable()`.

What are your thoughts on this?
![image](https://user-images.githubusercontent.com/59372212/117824935-2d890300-b26f-11eb-9db1-f21d71efc891.png)

Reply:
Better, though I still maintain that the "don't use `unwrap()`" advice applies to *all* Python API operations, since most of them can raise arbitrary exceptions, and feels out of place here.